src/MessageWriter.ts

```diff
@@ -203,6 +213,15 @@ export class MessageWriter {
           writer.sequenceLength(arrayLength);
         }
 
+        if (field.arrayLength != undefined && nestedMessage != undefined) {
```

Comment:
Should we also validate somewhere (maybe not here) that `fieldLength(defaultValue) === arrayLength`?

(Is it even possible to have a default value for fixed-length array fields?)

Reply:
I don't think that's necessary. ROS2 already ensures that during the build process:

```py
# msg def: fieldtype fieldname fielddefaultvalue
int32[5] five_integers_array [0, 0, 0, 0]
```

Error when building:
```
  rosidl_adapter.parser.InvalidValue: value '[0, 0, 0, 0]' can not be
  converted to type 'int32[5]': array must have exactly 5 elements, not 4
```